src/app/auth/forgotPassword/index.tsx

```diff
@@ -1,116 +1,91 @@
 import { router } from 'expo-router';
 import React, { useState } from 'react';
-import { Alert, TextInput, View } from 'react-native';
-import { Button, Input } from 'react-native-elements';
+import { Alert, Text, View } from 'react-native';
+import validator from 'validator';
 
 import styles from './styles';
 import globalStyles from '../../../styles/globalStyles';
 import { useSession } from '../../../utils/AuthContext';
+import UserStringInput from '../../../components/UserStringInput/UserStringInput';
+import StyledButton from '../../../components/StyledButton/StyledButton';
+import { isEmailTaken } from '../../../queries/profiles';
+import { queryEmailByUsername } from '../../../queries/auth';
 
 function ForgotPasswordScreen() {
   const { updateUser, signOut, resetPassword, verifyOtp } = useSession();
   const [email, setEmail] = useState('');
-  const [password, setPassword] = useState('');
-  const [loading, setLoading] = useState(false);
-  const [verificationCode, setCode] = useState<string>('');
-  const [changingPassword, setChangingPassword] = useState(false);
+  const [emailToReset, setEmailToReset] = useState('');
+  const [emailError, setEmailError] = useState('');
+  const [validEmail, setValidEmail] = useState(false);
 
   const sendResetEmail = async () => {
-    const { error } = await resetPassword(email);
+    const { error } = await resetPassword(emailToReset);
     if (error)
       Alert.alert('Could not send a reset password email. Please try again.');
-    else
-      Alert.alert(
-        'Enter the verification code from your email to change your password',
-      );
+    else router.replace('auth/signup');
   };
-  const verifyCode = async () => {
-    setLoading(true);
 
-    if (email && verificationCode) {
-      const { error } = await verifyOtp(email, verificationCode);
-
-      if (error) {
-        Alert.alert(error.message);
-        setChangingPassword(false);
-      } else {
-        setChangingPassword(true);
+  const setAndCheckEmailOrUsername = async (newEmail: string) => {
```

Comment:
So I'm seeing a really annoying bug when changing usernames really quickly, where valid usernames have the error showing. I uploaded a video so you can see what I mean.

https://github.com/calblueprint/girls-write-now/assets/34043950/a64071b0-3207-4321-b8dc-d48fc971215f

I think the problem is the onChange function of the textbox is still running when the text is being changed. Since we are querying the database, there can be random delays, so when you enter a valid username, the function is still being run on some old text in the text box and runs `setEmailError` after it is run on the most current text. 
The only way I can think of fixing is by making each call to the onChange function run one after another by using a mutex (please let me know if you have another idea). I was looking at this [library](https://www.npmjs.com/package/async-mutex) as something we could use. Since this sprint is getting more complicated, don't hesitate to ask me if you any questions with implementing this.